### PR TITLE
Fix f1_score calculation to not show NaN

### LINKED ISSUE #62: Fix bin_stats to not show f1 = NaN
When either precision or recall are 0, f1 is NaN instead of 0 as well

### PR DESCRIPTION
**Pull request by bvavrik**
Closes #62

### CODE REVIEW

**Comment by bvavrik** [comment]
Just to explain the upgrade:
The feature `f1_score = np.nan` when mask is array of zeros is kept
The added feature is that `f1_score = 0' when mask has some positive pixels, but prediction didn't hit any of them

File: cxflow_tensorflow/tests/metrics_test.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def test_dice(self):
         with tf.Session().as_default():
             labels = tf.constant(_LABELS, dtype=tf.int32)
             predictions = tf.constant(_PREDICTIONS, dtype=tf.int32)
-            expected_dice = np.array([(2*1.) / (3+2), 0])
+            expected_dice = np.array([(2*1.) / (3+2), 0, 0])
             computed_dice = bin_dice(predictions, labels).eval()
 
             self.assertTrue(np.all(expected_dice == computed_dice[:-1]))
```
Comment (petrbel) [inline cxflow_tensorflow/tests/metrics_test.py]: this should be replaced by `assertListEqual` (there are multiple occurrences)

File: cxflow_tensorflow/tests/metrics_test.py [hunk 2]
```diff
@@ -35,14 +35,18 @@ def test_stats(self):
...
             labels = tf.constant(_LABELS, dtype=tf.int32)
             predictions = tf.constant(_PREDICTIONS, dtype=tf.int32)
-            expected_recall = np.array([1./3, 0.])
+            expected_recall = np.array([1./3, 0., 0.])
             expected_precision = np.array([1./2, 0.])
-            expected_f1 = 2/(1/expected_precision + 1/expected_recall)
+            expected_f1 = np.array([0.4, 0.0, 0.0])
             computed_f1, computed_precision, computed_recall = \
                 [computed.eval() for computed in bin_stats(predictions, labels)]
+            print(expected_f1)
```
Comment (petrbel) [inline cxflow_tensorflow/tests/metrics_test.py]: tests should not contain `print`s

File: cxflow_tensorflow/tests/metrics_test.py [hunk 3]
```diff
@@ -35,14 +35,18 @@ def test_stats(self):
...
+            print(computed_f1)
+            print(type(expected_precision[:-1] == computed_precision[:-2]))
 
             self.assertTrue(np.all(expected_f1 == computed_f1[:-1]))
-            self.assertTrue(np.all(expected_precision == computed_precision[:-1]))
+            self.assertTrue(np.all(expected_precision == computed_precision[:-2]))
+            self.assertTrue(np.isnan(computed_precision[-2]))
```
Comment (petrbel) [inline cxflow_tensorflow/tests/metrics_test.py]: what about `[-1]`? it either should be nan or not, but tested in both alternatives

File: cxflow_tensorflow/tests/metrics_test.py (hunk 2 above)
Comment (bvavrik) [reply cxflow_tensorflow/tests/metrics_test.py]: Sorry, my bad

File: cxflow_tensorflow/tests/metrics_test.py (hunk 3 above)
Comment (bvavrik) [reply cxflow_tensorflow/tests/metrics_test.py]: is tested a bit later in 
`for computed in (computed_f1, computed_precision, computed_recall):`
`    self.assertTrue(np.isnan(computed[-1]))`

File: cxflow_tensorflow/tests/metrics_test.py (hunk 1 above)
Comment (bvavrik) [reply cxflow_tensorflow/tests/metrics_test.py]: I used the original formulation from pulled version

Comment (bvavrik) [reply cxflow_tensorflow/tests/metrics_test.py]: Those input are `np.ndarray`, so should I convert them to list and them compare, or leave it as it is?

**Pull request merged**